manifest/manifestlist/manifestlist_test.go

```diff
@@ -109,3 +110,105 @@ func TestManifestList(t *testing.T) {
 		}
 	}
 }
+
+// TODO (mikebrow): add annotations on the index and individual manifests
+var expectedOCIImageIndexSerialization = []byte(`{
```

Comment:
So:
* Yes, you would like to accept OCI indexes without platform
* You think the `platform` can be made optional for manifest lists (if I'm not mistaken, such manifests are accepted currently, but the result is `{ architecture: "", os: ""}`)
* Yes, you want it handled in this PR 

Making the platform truly optional looks like a two-liner - just make platform a pointer and mark it omitempty, then adjust the code that checks manifestDescriptor.Platform.Architecture in registry/handlers/manifests.go. However, there's similar code in moby/distribution that would also need modification. I don't know what the API maintenance policy is here.

A different approach would be to add a custom MarshalJSON to ManifestDescriptor that omits the platform if os, architecture, and other fields are all empty/missing. I don't know if this would be considered ugly :-)